pilot/cmd/pilot-agent/status/server.go

```diff
@@ -714,6 +728,66 @@ func (s *Server) handleAppProbeTCPSocket(w http.ResponseWriter, prober *Prober)
 	}
 }
 
+func (s *Server) handleAppProbeGRPC(w http.ResponseWriter, req *http.Request, prober *Prober) {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		// nolint: staticcheck
```

Comment:
I was using `grpc.WithInsecure`, the lint check showed:
```
pilot/cmd/pilot-agent/status/server.go:734:3: SA1019: grpc.WithInsecure is deprecated: use insecure.NewCredentials() instead. Will be supported throughout 1.x. (staticcheck)
		grpc.WithInsecure(), // credentials are currently not supported
		^
```

Then I changed to `insecure.NewCredentials()`, but it reports error:
```
pilot\cmd\pilot-agent\status\server.go:736:26: cannot use insecure.NewCredentials() (type credentials.TransportCredentials) as type grpc.DialOption in slice literal:
	credentials.TransportCredentials does not implement grpc.DialOption (missing grpc.apply method)
```

And [kubelet](https://github.com/kubernetes/kubernetes/blob/v1.23.1/pkg/probe/grpc/grpc.go#L58) also uses `grpc.WithInsecure()`

Reply:
Should be `grpc.WithTransportCredentials(insecure.NewCredentials())`. gRPC only uses it since they are on an older gRPC version I think, they are functionally equivilent.